src/Npgsql/PostgresEnvironment.cs

```diff
@@ -52,8 +52,10 @@ static class PostgresEnvironment
         }
 
         static string? GetDefaultFilePath(string fileName) =>
-            Environment.GetEnvironmentVariable(PGUtil.IsWindows ? "APPDATA" : "HOME") is string appData
-                ? Path.Combine(appData, "postgresql", fileName)
+            Environment.GetEnvironmentVariable(PGUtil.IsWindows ? "APPDATA" : "HOME") is string appData &&
+            Path.Combine(appData, "postgresql", fileName) is string filePath &&
```

Comment:
I like that expression, but it introduces an additional `null` check while a returned value from `Path.Combine' won't be `null` in our case. Does it matter?

Reply:
Definitely no issue with extra null check (this isn't perf-sensitive code). It's a bit weird because we're using a check to just assign a variable (basically the missing `let` in C#), but I'm OK with it.